cg/meta/demultiplex/demux_post_processing.py

```diff
@@ -107,23 +109,45 @@ def finish_flow_cell_temp(self, flow_cell_name: str) -> None:
             return
 
         # 3. Create flow cell.
-        try:
-            flow_cell: FlowCell = FlowCell(
-                flow_cell_path=flow_cell_dir, bcl_converter=bcl_converter
-            )
-        except FlowCellError:
-            LOG.error(f"Could not create flow cell for {flow_cell_name}")
+        parsed_flow_cell: Optional[FlowCell] = self.parse_flow_cell_data(
+            flow_cell_directory=flow_cell_dir,
+            bcl_converter=bcl_converter,
+        )
+
+        if not parsed_flow_cell:
             return
 
+        flow_cell: Flowcell = self.create_flow_cell(parsed_flow_cell=parsed_flow_cell)
+
         # 4. Store flow cell in status db.
         self.status_db.session.add(flow_cell)
         self.status_db.session.commit()
 
-        # 5. Store flow cell data in housekeeper. TODO
+        # 5. Store flow cell data in housekeeper.
 
         # 6. Create sequencing metrics
         self.add_sample_lane_sequencing_metrics_for_flow_cell(flow_cell_name=flow_cell_name)
 
+    def create_flow_cell(self, parsed_flow_cell: FlowCell) -> Flowcell:
+        """Create flow cell from the parsed and validated flow cell data."""
+        return Flowcell(
+            name=parsed_flow_cell.id,
+            sequencer_type=parsed_flow_cell.sequencer_type,
+            sequencer_name=parsed_flow_cell.machine_name,
+            sequenced_at=parsed_flow_cell.run_date,
+        )
+
+    def parse_flow_cell_data(self, flow_cell_directory: Path, bcl_converter: str) -> FlowCell:
```

Comment:
Could this not just be:
```suggestion
    def get_flow_cell(self, flow_cell_directory: Path, bcl_converter: str) -> FlowCell:
```

Reply:
Yes, that might be better.
I think `parse_flow_cell_data`  is a good name because it reads the directory structure and returns the data in it